src/posts/xarray-biology/index.md

```diff
@@ -0,0 +1,157 @@
+
+### Who is Using Xarray
+
+- [`Sgkit`](https://sgkit-dev.github.io/sgkit/latest/getting_started.html#data-structures) "a toolkit for quantitative and population genetics" uses `Xarray` `Dataset`'s as a core data structure. `Xarray` allows them to keep track of the rich metadata associated with sequences and further uses computational patterns to speed up analyses.
+  - `Xarray`'s interoperability with Zarr makes it a promising way to interact with new Variant Call Format (VCF) standard [`VCF Zarr`](https://www.biorxiv.org/content/10.1101/2024.06.11.598241v3)
+- `Xarray` has also been used for neurophysiology data by the [Ecephys](https://allensdk.readthedocs.io/en/latest/allensdk.brain_observatory.ecephys.html) package. See a previous blog post [xarray for neurophysiology](https://xarray.dev/blog/xarray-for-neurophysiology) to learn more.
+- The `scverse` family of software uses `anndata`, which, like `Xarray` provides labeled dimensions. There are plans to have `anndata` be powered by `anndata`
+- The `bioio` package, capable of reading a zoo of microscope formats, provides a function to get image stacks as a `DataArray`.
+- The `Xarray.DataTree` provides a natural representation of multiscale images and can interoperate well with the OME-NGFF model. The [multiscale-spatial-image](https://github.com/spatial-image/multiscale-spatial-image?tab=readme-ov-file#multiscale-spatial-image) project uses `DataTree` for precisely this purpose.
+
+## What has limited adoption by Biologists?
+
+Given the benefits of switching to `Xarray`, why aren't more biologists using it? Is it secretly not as good as this blog claims? I spent February and April 2025 reaching out to scientists and scientific software developers across various fields of biology to find out. From those discussions, I've found three core reasons have prevented biologists who are already using Python from adopting `Xarray`.
+
+### Awareness and Examples
+
+Many biologists have never heard of `Xarray`. This is because the geosciences community created Xarray, and there wasn't enough community overlap for the word to spread to biologists.
+
+A related issue is that the examples and tutorials are geoscience-centric, which was the source of my discouragement when I first read the Xarray docs.
+
+### Technical Barriers
+
+Once a potential user is convinced of `Xarray`'s value, they may still face technical barriers. Ranging from rough edges to missing features, however, none are insurmountable. These barriers can exist in `Xarray` or in other tools in their workflow not accepting `Xarray` as input.
+
+An example of a rough edge in `Xarray` is that, as of May 2025, you cannot use integers as keys in a `DataTree`. That is a problem, as integers are a natural key when tracking single-cell lineages. Rough edges like this one haven't been smoothed over yet because there has not been a large user base of biologists using `Xarray` and raising issues when they encounter problems.
```

Comment:
```
An example of a rough edge in `Xarray` is that you cannot use integers as keys in a `DataTree`. That is a problem, as integers are a natural key when tracking single-cell lineages. Rough edges like this one haven't been smoothed over yet because there has not been a large user base of biologists raising issues when they encounter problems.
```